src/common/payment.ts

```diff
@@ -207,4 +269,110 @@ export class TurboAuthenticatedPaymentService
       await this.signer.generateSignedRequestHeaders(),
     );
   }
+
+  private async getTargetWalletForFund(): Promise<string> {
+    const { addresses } = await this.httpService.get<TurboInfoResponse>({
+      endpoint: '/info',
+    });
+
+    return addresses[this.token];
+  }
+
+  // TODO: token based backoff
+  private async pollForTxBeingAvailableFromGateway(txId: string) {
+    const maxAttempts = 30;
+    const pollingIntervalMs = 1_000;
+    const initialBackoffMs = 5_000;
+
+    await new Promise((resolve) => setTimeout(resolve, initialBackoffMs));
+
+    let attempts = 0;
+    while (attempts < maxAttempts) {
+      const response = await this.arweave.api.post('/graphql', {
+        query: `
+          query {
+            transaction(id: "${txId}") {
+              recipient
+              owner {
+                address
+              }
+              quantity {
+                winston
+              }
+            }
+          }
+        `,
+      });
+
+      const transaction = response.data.data.transaction;
+
+      if (transaction) {
+        return;
+      }
+      attempts++;
+      this.logger.debug('Transaction not found after polling...', {
+        txId,
+        attempts,
+      });
+      await new Promise((resolve) => setTimeout(resolve, pollingIntervalMs));
+    }
+
+    throw new Error('Transaction not found after polling');
+  }
+
+  public async fund(
+    tokenAmount: BigNumber.Value,
+    feeMultiplier = 1,
```

Comment:
nit, use an object

Reply:
On these fund and submitFundTransaction, I added methods that have the same API as irysSDK.fund/submitFundTx

Reply:
switched to object oriented, and re named `fund` to `fundWithTokens` as we have existing interfaces for fiat